### PR TITLE
IA-1947 handle bulk import with task worker

### PR DESCRIPTION
**Pull request by SimonDus**
The bulk create user has been moved as a task. To do it I moved the code that actually create the users out of the view and made it a task. Tests are covering the user creation part. 

Related JIRA tickets : IA-1873

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new string have been introduced in the frontend
- [x] My migrations file are included
- [x] Are there enough tests
- [x] Documentation has been included (for new feature)

## How to test

make a csv file to create users. Check the files from the tests to find how to do it. 
go to /api/bulkcreateuser/ and upload your file. 
Go to task and check if the task is created. 

For the user creation itself,  you must run it from iaso bash using :

` bulk_create_users_task.__wrapped__(uploaded_file, user, False)`

uploaded_file is your csv file mock as an upload.
user is your logged in user
False is to prevent triggering task log when not performing it as a task.

Dont forget to wrap it to dont trigger the task. 

The file must be a mock of an uploade file so you must do it this way : 

```
file = open("path_to_your_test_file")

request = RequestFactory().post("/upload", {"file": file})
uploaded_file = request.FILES["file"]
```

### CODE REVIEW

File: hat/common/utils.py [hunk 1]
```diff
@@ -20,3 +22,11 @@ def queryset_iterator(queryset, chunk_size=1000):
             yield from paginator.get_page(index + 1)
     else:
         yield from queryset.iterator(chunk_size=chunk_size)
+
+
+def is_json_serializable(x):
```
Comment (mathvdh) [inline hat/common/utils.py:27]: Ce check additionnel sur les paramètres via `is_json_serializable` est intéressant.

Néanmoins dans l'état il me semble que ça peut péter, en effet dans la doc du module json ils parent d'autres erreurs, telles que `RecursionError` ou `ValueError`

Du coup est-ce que tu ne ferais pas un except vide pour être sur que ça throw pas s'il se passe quoi que ce soit d'autre dans le json.dumps ?

File: iaso/api/bulk_create_users.py [hunk 2]
```diff
@@ -15,7 +15,175 @@
 from rest_framework.decorators import action
 from django.http import FileResponse
 
-from iaso.models import BulkCreateUserCsvFile, Profile, OrgUnit
+from beanstalk_worker import task_decorator
+from iaso.api.tasks import TaskSerializer
+from iaso.models import BulkCreateUserCsvFile, Profile, OrgUnit, ERRORED
+
+
+@task_decorator(task_name="bulk_create_users")
```
Comment (mathvdh) [inline iaso/api/bulk_create_users.py]: Il me semble que le code des tasks est un fichier spécifique dans le repertoire `/iaso/tasks` ? Et pas directement dans l'API ... Mais je peux me tromper .. ?

**Review by mathvdh — CHANGES_REQUESTED** [review]

File: hat/common/utils.py (hunk 1 above)
Comment (SimonDus) [reply hat/common/utils.py:27]: Pas faux

File: iaso/api/bulk_create_users.py (hunk 2 above)
Comment (SimonDus) [reply iaso/api/bulk_create_users.py]: Je bouge ça

**Review by mathvdh — APPROVED** [review]
👍🏼

**Comment by SimonDus** [comment]
I found a bug. Don't merge it for now

**Comment by SimonDus** [comment]
I close this PR and will start again from main as it's too outdated and a mess to merge

**Pull request closed without merging**